### PR TITLE
KV: Update 'versioned' naming to 'v2'

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: vault/logical_system.go [hunk 1]
```diff
@@ -1665,13 +1665,22 @@ func (b *SystemBackend) handleMount(ctx context.Context, req *logical.Request, d
 		config.PassthroughRequestHeaders = apiConfig.PassthroughRequestHeaders
 	}
 
-	// Alias versioned KV
-	if logicalType == "vkv" {
+	// Alias KV v1
+	if logicalType == "kv-v1" {
```
Comment (calvn) [inline vault/logical_system.go]: Can these alias checks be moved into the switch in https://github.com/hashicorp/vault/pull/4293/files#diff-a145099c8e3d917858619dcfb2ae09b9L1627?

**Comment by jefferai** [comment]
Looks good -- I'm guessing mount tune is fine without changes since you specify the option directly, but it might be nice to add a version flag for parity. Other than that LGTM.

File: command/auth_enable.go [hunk 2]
```diff
@@ -160,6 +162,13 @@ func (c *AuthEnableCommand) Flags() *FlagSets {
 		Usage:   "Enable seal wrapping of critical values in the secrets engine.",
 	})
 
+	f.IntVar(&IntVar{
+		Name:    "version",
+		Target:  &c.flagVersion,
+		Default: 0,
```
Comment (meirish) [inline command/auth_enable.go]: So the version will be `0` for "old" kvs, and `2` for the new one?

Comment (jefferai) [reply command/auth_enable.go]: I think those will be interchangeable, with zero being interpreted as "not set"

Comment (briankassouf) [reply command/auth_enable.go]: Yep! What jeff said

**Pull request merged**